account_move_import/wizard/import_move_line_wizard.py

```diff
@@ -0,0 +1,330 @@
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import base64
+import StringIO
+import csv
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+import logging
+_logger = logging.getLogger(__name__)
+
+header_fields = [
+    'account', 'debit', 'credit', 'name', 'partner', 'date_maturity',
+    'amount_currency', 'currency', 'tax_code', 'tax_amount',
+    'analytic_account']
+
+
+class aml_import(orm.TransientModel):
+    _name = 'aml.import'
+    _description = 'Import account move lines'
+    _columns = {
+        'aml_data': fields.binary('File', required=True),
+        'aml_fname': fields.char('Filename', size=128, required=True),
+        'csv_separator': fields.selection(
+            [(',', ','), (';', ';')], 'CSV Separator', required=True),
+        'decimal_separator': fields.selection(
+            [('.', '.'), (',', ',')], 'Decimal Separator', required=True),
+        'note': fields.text('Log'),
+    }
+    _defaults = {
+        'aml_fname': '',
+    }
+
+    def aml_import(self, cr, uid, ids, context=None):
+
+        account_obj = self.pool.get('account.account')
+        move_obj = self.pool.get('account.move')
+        partner_obj = self.pool.get('res.partner')
+        curr_obj = self.pool.get('res.currency')
+        tax_code_obj = self.pool.get('account.tax.code')
+        analytic_obj = self.pool.get('account.analytic.account')
+        mod_obj = self.pool.get('ir.model.data')
+        dp = self.pool.get('decimal.precision').precision_get(
+            cr, uid, 'Account')
+
+        company_id = context['company_id']
+        move_id = context['move_id']
+        account_ids = account_obj.search(
+            cr, uid, [('type', '!=', 'view'), ('company_id', '=', company_id)])
+        accounts = account_obj.read(cr, uid, account_ids, ['code', 'name'])
+
+        result_view = mod_obj.get_object_reference(
+            cr, uid, 'account_move_import', 'aml_import_result_view')
+        data = self.browse(cr, uid, ids)[0]
+        aml_file = data.aml_data
+        csv_separator = str(data.csv_separator)
+        decimal_separator = data.decimal_separator
+
+        err_log = ''
+        header_line = False
+
+        lines = base64.decodestring(aml_file)
+        reader = csv.reader(
+            StringIO.StringIO(lines), delimiter=csv_separator)
+        sum_debit = sum_credit = 0.0
+        amls = []
+
+        for ln in reader:
+
+            if not ln or ln and ln[0] and ln[0][0] in ['', '#']:
+                continue
+
+            # process header line
+            if not header_line:
+                if ln[0].strip().lower() not in header_fields:
+                    raise orm.except_orm(
+                        _('Error :'),
+                        _("Error while processing the header line %s."
+                          "\n\nPlease check your CSV separator as well as "
```

Comment:
@luc-demeyer Using csv.Sniffer would eliminate this error case. Users can make mistakes, Python not.
This would simplifiy header processing logic and make it easer to extract as a method later.

Regards